### ISSUE TITLE
Copy/paste of two-bar repeat produces empty measures

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#21634]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

When using copy/paste on a passage containing a two-bar repeat, the repeat is not copied.

### Steps to reproduce

1. add a two-bar repeat to a score
2. select those two bars or any range containing them
3. copy
4. select a destination
5. paste

Result: the bars corresponding to the two-bar repeat as empty in the destination

Same for four-bar repeat.

### Screenshots/Screen recordings

![image](https://github.com/musescore/MuseScore/assets/1799936/a75030ec-7ebb-426f-a96c-25d3e4ca96dc)

### MuseScore Version

OS: Debian GNU/Linux 11 (bullseye), Arch.: x86_64, MuseScore version (64-bit): 4.3.0-240530518, revision: github-musescore-musescore-0fa51e6

### Regression

I don't know

### Operating system

Debian 11

### Additional context

Same in 4.2.1 and 4.1.1; I can't easily test earlier MU4 releases.  Not applicable to MU3 since there was no two-bar repeat.

I thought I remembered seeing an issue about this previously but am not finding it, open or closed.  Since these elements also involve "no break" elements, it's no doubt a bit complicated to make this work.  We could consider putting up a warning dialog, much as we do for "tuplet would cross barline" - but I'd recommending allowing the copy anyhow so the user can simply add back the repeats.

**Comment by bkunda** [comment]
@oktophonie pls advise whether or not this is something to be fixed on the engraving side, and we can assign a dev accordingly.

**Comment by MarcSabatella** [comment]
I'd be pretty sure it's more a matter of full handling for the multibar repeat never being added to the code that implements the paste command - for instance here:

https://github.com/musescore/MuseScore/blob/2ba38f4e396f3f8d7cca90d516df9d08d4d65e60/src/engraving/rw/read400/read400.cpp#L334

Looks like the repeat group is simply deleted, and it was that way from when this code was first added.  I mentored the GSoC project where this was implemented but have no recollection of whether copy/paste was ever known to be working.

**Comment by MarcSabatella** [comment]
Indeed, this *was* mentioned specifically in the final "work product" for the project - see https://musescore.org/en/user/101731/blog/2020/08/28/gsoc-2020-work-product-new-and-improved-multimeasure-rests-and-measure:

> Pasting—while two-measure and four-measure repeats can be copied to the clipboard, at present when pasted they will be replaced by rests. Although in many situations the user would desire measure repeats to be pasted as they were copied, in other scenarios, such as when pasting onto a different beat than the selection originally started on, this is the most natural behavior. It is a result of code to automatically delete an existing measure repeat when any content is pasted that would conflict with it.

**Comment by skdonnelly** [comment]
This makes it very tedious to copy/paste repetitive drum tracks. Hoping for a fix, even if it's only when pasting full bars. Oddly you can select and copy the 2-bar rest symbol itself, then select one or two empty bars and hit paste and it does paste the 2-bar rest properly into the two bars, but copying full bars that contain the symbol doesn't work. Also if I select a series of notes and the 2-bar rest, then copy, when I paste into another staff it pastes *only* the 2-bar rest, not the notes.

**Comment by MarcSabatella** [comment]
In theory, you shouldn't normally need to use copy/paste when filling a passage with two-bar repeats - just select the entire passage first before clicking the palette item.  Except that doesn't work.  It works for one-bar repeats but not two.  Which is also probably a pretty simple enhancement.

**Comment by skdonnelly** [comment]
Yes but I like to have the drum part or whatever shown every 8 bars, then have the 2-bar repeats for the other 6 bars of the 8, to have reminders so the person doesn't fall asleep. :) Or so that their part is repeated on subsequent pages as a reminder. You're right about clicking on a symbol with multiple bars selected, it fills all the bars for almost all the symbols in the Repeats and jumps palette, 2-bar and 4-bar repeats being the exception. They should just fill what they can, and if there are less bars remaining than the number of repeats, they should just leave the remainder as is.
